esx_service/utils/auth_data.py

```diff
@@ -229,7 +235,7 @@ def replace_vms(self, conn, vms):
             )
 
             conn.executemany(
-                "INSERT INTO vms(vm_id, tenant_id) VALUES (?, ?)",
+                "INSERT INTO vms(vm_id, vm_name, tenant_id) VALUES (?, ?, ?)",
```

Comment:
This replaces the existing VMs that belong to a tenant with the new list of VMs.
Single INSERT OR REPLACE could work; but it again require removal of records of earlier VMs that don't belong to tenant (or update their tenant_id to empty).